test/unit-test/core_mqtt_serializer_utest.c

```diff
@@ -1401,6 +1401,13 @@ void test_MQTT_DeserializeAck_connack( void )
     buffer[ 1 ] = 1;
     status = MQTT_DeserializeAck( &mqttPacketInfo, &packetIdentifier, &sessionPresent );
     TEST_ASSERT_EQUAL_INT( MQTTBadResponse, status );
+    TEST_ASSERT_EQUAL_INT( true, sessionPresent );
```

Comment:
I don't think it really makes sense to assert on this since this test case tests an error path

Reply:
Yes, I intended to remove it. Removed.